### PR TITLE
Add new custom ES lint rule to error on usage of modifying style prop…

### PR DESCRIPTION
**Pull request by alabiaga**
Throw an ES lint violation when modifying the styles of elements. See #14424 

This custom rule will catch the following code and throw an error

element.style.some-property = ...
element.style['some-property'] = 
element.style[propertyName] =

and whitelists the usage of style property mutations in style.js, fixed-layer.js, and in tests.

Sample run:
>> gulp lint
.../amphtml/extensions/amp-story/0.1/amp-story-grid-layer.js
  29:1  error  .../amphtml/extensions/amp-story/0.1/amp-story-grid-layer.jsThe setting of a style property is forbidden. Please use style.js#setStyles or style.js#setImportantStyles  no-style-property-setting

Note that this will not be checked in until PR #14424 is checked in as that PR fixes amp-story-grid-layer.js usage of style.

Note that this doesn't catch the following

element['style']['some-property] = ...

I don't think that particular coding style is too common but if anyone feels strongly about adding this, let me know.

### CODE REVIEW

**Comment by alabiaga** [comment]
\cc @choumx

**Comment by alabiaga** [comment]
\cc @rsimha

File: build-system/eslint-rules/no-style-property-setting.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+module.exports = function(context) {
+  const sourceCode = context.getSourceCode();
+  return {
+    Program: function(node) {
+      // Ignore style.js, fixed-layer.js and tests for usage of style property
+      // changes.
+      if (!/^(style|fixed-layer|test-(\w|-)+)\.js$/.test(
+            path.basename(context.getFilename()))) {
+        const sourceCodeAsText =
```
Comment (jridgewell) [inline build-system/eslint-rules/no-style-property-setting.js]: The power of ESLint comes from the AST, let's take advantage of that:

```js
export default function(context) {
  return {
    MemberExpression(node) {
      // Ignore style.js, fixed-layer.js and tests for usage of style property
      // changes.
      const filename = context.getFilename();
      if (/test-/.test(filename)) {
        return;
      }
      if (/src\/(style|fixed-layer)\.js$/.test(filename)) {
        return;
      }
      if (node.computed) {
        return;
      }
      
      if (node.property.name === "style") {
        context.report(node, 'The use of Element#style (CSSStyleDeclaration live object) to style elements is forbidden. Use getStyle and setStyle from style.js');
      }
    },
  };
};
```

Comment (alabiaga) [reply build-system/eslint-rules/no-style-property-setting.js]: ah this is more concise. I'll make these changes. Thanks Justin.

Comment (alabiaga) [reply build-system/eslint-rules/no-style-property-setting.js]: changes made. please review. thanks

**Comment by rsimha** [comment]
@alabiaga I think the Travis build failed due to a stale cache of `node_modules/eslint-plugin-amphtml-internal`.

Ideally, we should be able to exclude the directory from the cache, but Travis doesn't provide a way to do so right now. (https://github.com/travis-ci/travis-ci/issues/3039)

You could add a call to `yarn add file:build-system/eslint-rules` to `.travis.yml`, but that will be wasteful most of the time.

I've cleared the Travis cache so you can re-run the failing shard on your Travis build.

**Comment by choumx** [comment]
Should we remove the obviated presubmit check?

**Comment by alabiaga** [comment]
@rsimha sounds good. Will some tests that are broken due to the new rule and I will also make that change. Thanks

File: build-system/eslint-rules/no-style-property-setting.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ * limitations under the License.
+ */
+'use strict';
+
+const path = require('path');
+
+module.exports = function(context) {
+  const ignoredFiles = new RegExp(
+      '^(keyframes-extractor|fixed-layer|style|test-(\\w|-)+'
+      + '|(\\w|-)+-testing)\\.js$');
```
Comment (jridgewell) [inline build-system/eslint-rules/no-style-property-setting.js]: Instead of extracting, just split these into multiple checks.

**Comment by rsimha** [comment]
Turns out that incrementing the version number doesn't do the trick on Travis. @jridgewell discovered that `yarn link` might work instead.

File: build-system/eslint-rules/package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "eslint-plugin-amphtml-internal",
-  "version": "0.1.0",
```
Comment (jridgewell) [inline build-system/eslint-rules/package.json]: Revert this, it doesn't work on Travis. Verified with @rsimha

**Comment by rsimha** [comment]
Tested locally with @alabiaga after rebasing on #14674. This is good to merge after Travis goes green.

**Pull request merged**